istioctl/pkg/validate/validate.go

```diff
@@ -158,6 +163,39 @@ func (v *validator) validateResource(istioNamespace string, un *unstructured.Uns
 		v.validateDeploymentLabel(istioNamespace, un)
 		return nil
 	}
+
+	if un.GetAPIVersion() == "install.istio.io/v1alpha1" {
+		if un.GetKind() == "IstioOperator" {
+			if err := checkFields(un); err != nil {
+				return err
+			}
+
+			// IstioOperator isn't part of pkg/config/schema/collections,
+			// usual conversion not available.
+
+			spec, ok := un.Object["spec"]
```

Comment:
could we move this logic into the operator validate package and create a function there that takes an IstioOperator string? this logic seems too low level/IOP specific to be in this function.

Reply:
Done